File: palindrome_checker.py
def is_palindrome(s):
    # Remove spaces and convert to lowercase
    cleaned_string = ''.join(s.split()).lower()
    # Check if the string is the same forwards and backwards
    return cleaned_string == cleaned_string[::-1]

def main():
    print("Welcome to the Palindrome Checker!")
    
    # Get user input
    user_input = input("Enter a string to check if it's a palindrome: ")
    
    # Check if the input is a palindrome
    if is_palindrome(user_input):
        print(f'"{user_input}" is a palindrome.')
    else:
        print(f'"{user_input}" is not a palindrome.')

if __name__ == "__main__":
    main()

File: palindrome_checker.php
<?php

function isPalindrome($string) {
    // Convert the string to lowercase and remove non-alphanumeric characters
    $cleanedString = preg_replace("/[^A-Za-z0-9]/", '', strtolower($string));

    // Check if the cleaned string is the same as its reverse
    return $cleanedString === strrev($cleanedString);
}

// Example usage
$inputString = "A man, a plan, a canal, Panama";
if (isPalindrome($inputString)) {
    echo "'$inputString' is a palindrome.";
} else {
    echo "'$inputString' is not a palindrome.";
}

?>
